### PR TITLE
Quizzes!!!

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?
You wanted quizzes, you got em', @carriewright11 Here's a bunch of quizzes for the chapters that were already merged. I will turn on the Leanpub quiz formatting too so once we figure out how to publish on Leanpub through the Hutch we are ready to blast these off! 

#### What was your approach?

I rapid fire made a bunch of multiple choice quiz questions for the chapters that are already merged. I used ChatGPT to give me drafts which I edited and made Leanpub Markua format.

### CODE REVIEW

File: quizzes/quiz_ch2.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{quiz, id: reproducible_code, attempts: 10}
+
+## What is R?
+
+? What is R?
+
+A) A language for working with data
+b) A set of code that allows you to do various things
+c) A type of programming notation
+d) A language for creating websites
```
Comment (carriewright11) [inline quizzes/quiz_ch2.md]:
```suggestion
d) A language specifically for creating websites
```

File: quizzes/quiz_ch2.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+b) A way to protect your code from being seen by others
+c) A type of statistical analysis
+D) A set of code that can be shared between users
+
+? What is one benefit of having a rich and active community in R?
+a) More opportunities to make your code more complicated
+b) Less support for users
+C) More resources for finding help and tutorials
+d) More difficult to find others who use R
+
```
Comment (carriewright11) [inline quizzes/quiz_ch2.md]: Maybe we could add questions about why R is good for reproducibility?

File: quizzes/quiz_ch3.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+? Which of the following is not a component of reproducibility?
+
+a) Transparency
+b) Accessibility
+c) Consistency
+D) Efficiency
+
+? What does "open source" mean in the context of reproducibility?
+
+A)  Making data and code available for anyone to access and modify
```
Comment (carriewright11) [inline quizzes/quiz_ch3.md]:
```suggestion
A) Making data and code available for anyone to access and suggest modifications
```

Comment (carriewright11) [reply quizzes/quiz_ch3.md]: "and modify" sounds like random people might modify your code - maybe we mean to access and use? just wondering if we could phrase it a bit differently. I know you mean code review, but just thinking for people new to the subject that this might sound concerning

File: quizzes/quiz_ch3.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+a) It makes the analysis more complex and difficult to execute
+b) It helps others to be able to publish conclusions about our data
+C) It helps others be able to truly reproduce the analysis by using the same data
+d) It exposes personal identifiable information to the public
+
+
+? True or False - if your data has PHI or PII you cannot make your analysis open source or reproducible.
+
+a) TRUE - There is no methods that will allow you to share your methods and data in a safe way.
+B) FALSE! - These are not mutually exclusive goals! but will take a bit of thoughtful planning and using controlled access methods so that all individuals' data are well protected!
```
Comment (carriewright11) [inline quizzes/quiz_ch3.md]:
```suggestion
B) FALSE! - These are not mutually exclusive goals! Some thoughtful planning and use of controlled access methods can ensure that all individuals' data are well protected!
```

File: quizzes/quiz_ch7.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+b) An online platform for sharing code.
+c) A way to experiment with different code without affecting the main analysis.
+d) A method for collaborating on the same code simultaneously.
+
+? What is the benefit of using GitHub for sharing code?
+
+a) It allows you to store your code in a way that obscures it from others.
+b) It provides a way to publish your research in a peer reviewed manuscript
+C) It allows you to track your code and project files as you and others continue to work on them.
+d) It makes it easy direct message different team members. .
```
Comment (carriewright11) [inline quizzes/quiz_ch7.md]:
```suggestion
d) It makes it easy direct message different team members.
```

File: quizzes/quiz_ch7.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+## GitHub Workflow
+
+? What is version control?
+
+A) A method of tracking files as they are changed throughout a project.
+b) An online platform for sharing code.
+c) A way to experiment with different code without affecting the main analysis.
+d) A method for collaborating on the same code simultaneously.
+
+? What is the benefit of using GitHub for sharing code?
```
Comment (carriewright11) [inline quizzes/quiz_ch7.md]:
```suggestion
? What is the major benefit of using GitHub for sharing code?
```

File: quizzes/quiz_ch7.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+a) It allows you to store your code in a way that obscures it from others.
+b) It provides a way to publish your research in a peer reviewed manuscript
```
Comment (carriewright11) [inline quizzes/quiz_ch7.md]:
```suggestion
b) It provides a way to publish your research in a peer reviewed manuscript.
```

File: quizzes/quiz_ch8.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+a) A type of operating system
+B) All the relevant pieces of software and their dependencies that were used on a computer at the time that an analysis or other project was run
+c) The physical components of a computer
+d) The programming language used to run an analysis
+
+? What is "dependency hell"?
+
+A) The feeling of frustration when trying to update software that exists in a tangled mess of software dependencies
+b) The name of a popular computer game
```
Comment (carriewright11) [inline quizzes/quiz_ch8.md]: :P

File: quizzes/quiz_ch8.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+c) The process of trying to find software dependencies
```
Comment (carriewright11) [inline quizzes/quiz_ch8.md]: this answer seems kinda true too... maybe rephrase?

File: quizzes/quiz_ch8.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+? Why is it important to control for software versions in creating reproducible analyses?
+
+a) Because it makes the analysis run faster
+b) Because it makes the analysis easier to understand
+C) Because software versions can directly affect not only whether an analysis will be able to run, but the results of the analysis
+d) Because it ensures that the analysis is correct
+
+? What is session info?
+
+A) A printout in R that displays information about the software and packages that were being used at the time the sessionInfo() function was run.
```
Comment (carriewright11) [inline quizzes/quiz_ch8.md]:
```suggestion
A) A printout in R that displays information about the software and packages that were being used at the time the sessionInfo() function was run
```

Comment (carriewright11) [reply quizzes/quiz_ch8.md]: making it match the other answers

**Comment by carriewright11** [comment]
woohoo way to get these done quick @cansavvy!

File: quizzes/quiz_ch8.md (hunk 9 above)
Comment (cansavvy) [reply quizzes/quiz_ch8.md]: Will edit phrasing

Comment (cansavvy) [inline quizzes/quiz_ch8.md]:
```suggestion
c) The process of trying to find software that is useful
```

File: quizzes/quiz_ch3.md (hunk 3 above)
Comment (cansavvy) [reply quizzes/quiz_ch3.md]: How's this? @carriewright11

File: quizzes/quiz_ch2.md (hunk 2 above)
Comment (cansavvy) [reply quizzes/quiz_ch2.md]: Good call

File: quizzes/quiz_ch2.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (cansavvy) [inline quizzes/quiz_ch2.md]:
```suggestion

? Why is R useful for reproducibility in analyses? 
a) It is free which means pricing is not a barrier to reproducing results with R.
b) It has an active community who can contribute and review the reproducibility of it.
c) It has an active developer community who has made lots of handy tools which aid reproducibility. 
D) All of the above
```

Comment (cansavvy) [reply quizzes/quiz_ch2.md]: How's this? @carriewright11

File: quizzes/quiz_ch3.md (hunk 3 above)
Comment (carriewright11) [reply quizzes/quiz_ch3.md]: looks good!

File: quizzes/quiz_ch2.md (hunk 11 above)
Comment (carriewright11) [reply quizzes/quiz_ch2.md]: looks good!

**Comment by cansavvy** [comment]
There's no actual quiz errors. It thinks that by the report printing out `NA` that means there's an error. I tracked this as an issue in OTTR: https://github.com/jhudsl/OTTR_Template/issues/635 

So these quizzes are ready to go as far as I'm concerned, @carriewright11

**Comment by carriewright11** [comment]
> There's no actual quiz errors. It thinks that by the report printing out `NA` that means there's an error. I tracked this as an issue in OTTR: [jhudsl/OTTR_Template#635](https://github.com/jhudsl/OTTR_Template/issues/635)
> 
> So these quizzes are ready to go as far as I'm concerned, @carriewright11

oooh ok good to know
